public/js/controllers/charts_controller.js

```diff
@@ -605,9 +669,20 @@ export default class extends Controller {
   }
 
   _zoomCallback (start, end) {
+    let currentSettings = {}
+    if (this.isSpecialXAxis()) {
+      this.settings.limit = this.selectedZoom()
+      currentSettings = this.settings
+      this.query.nullEntry(currentSettings, ['zoom', 'axis', 'bin', 'scale'])
```

Comment:
previous two line are the same as `this.query.nullEntry(this.settings, ['zoom', 'axis', 'bin', 'scale'])`

In javascript, all objects and arrays are assigned by reference.